### PR TITLE
Reverse hostname on Trie entries

### PR DESCRIPTION
**Pull request by cvazac**
byte savings when comparing `encodeURIComponent(JSON.stringify(trie)).length`:

- https://news.google.com/ (221, 12%)
- http://www.msn.com/ (208, 4%)
- http://www.sfgate.com/ (3058, 12%)

### CODE REVIEW

File: src/resourcetiming-compression.js [hunk 1]
```diff
@@ -803,6 +804,23 @@
         return this.optimizeTrie(this.convertToTrie(results), true);
     };
 
+    // eslint-disable-next-line max-len
+    ResourceTimingCompression.HOSTNAME_REGEX = /^(?:(?:(([^:\/#\?]+:)?(?:(?:\/\/)(?:(?:(?:([^:@\/#\?]+)(?:\:([^:@\/#\?]*))?)@)?(([^:\/#\?\]\[]+|\[[^\/\]@#?]+\])(?:\:([0-9]+))?))?)?)?((?:\/?(?:[^\/\?#]+\/+)*)(?:[^\?#]*)))?(\?[^#]+)?)(#.*)?/;
```
Comment (nicjansma) [inline src/resourcetiming-compression.js]: All constants should be up at the top.

This regex is insane!  It looks like this is breaking up the entire URL, so you get capture groups for everything:

image

Can we get a simpler regex that just matches the domain?

```
https?:\/\/([^\/]+)
```

image

File: src/resourcetiming-compression.js [hunk 2]
```diff
@@ -803,6 +804,23 @@
...
+    ResourceTimingCompression.reverseHostname = function(url) {
```
Comment (nicjansma) [inline src/resourcetiming-compression.js]: JSDoc comments please

File: src/resourcetiming-compression.js [hunk 3]
```diff
@@ -803,6 +804,23 @@
...
+        var hostname = (ResourceTimingCompression.HOSTNAME_REGEX.exec(url) || [])[5];
+        return typeof hostname === "string"
+            ? url.split(hostname).join(ResourceTimingCompression.reverseString(hostname))
+            : url;
+    };
+
+    ResourceTimingCompression.reverseString = function(i) {
+        var l = i.length, o = "";
```
Comment (nicjansma) [inline src/resourcetiming-compression.js]: Not that we have to hyper-optimize this, but I came across this test:
https://jsperf.com/reverse-string

The split/reverse/join seems to be fastest overall (though I don't think we should update the `String.prototype`)

**Comment by nicjansma** [comment]
Also, do you have any tests available?  Do any tests need to be updated?

File: src/resourcetiming-compression.js (hunk 3 above)
Comment (cvazac) [inline src/resourcetiming-compression.js]: ah, but i've already hyper-optimized! ;)
https://jsperf.com/reverse-string2

File: src/resourcetiming-decompression.js [hunk 4]
```diff
@@ -390,7 +394,7 @@
             split.push(split[0]);
         }
 
-        if (split.length === 2) {
+        if (split.length === 2 && true) {
```
Comment (nicjansma) [inline src/resourcetiming-decompression.js]: Probably not needed? :)

**Pull request merged**